subsys/bluetooth/mesh/pb_gatt_cli.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2021 Xiaomi Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <sys/byteorder.h>
+
+#include <net/buf.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/hci.h>
+#include <bluetooth/uuid.h>
+#include <bluetooth/conn.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/mesh.h>
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_PROV)
+#define LOG_MODULE_NAME bt_mesh_pb_gatt_client
+#include "common/log.h"
+
+#include "mesh.h"
+#include "adv.h"
+#include "net.h"
+#include "rpl.h"
+#include "transport.h"
+#include "host/ecc.h"
+#include "prov.h"
+#include "pb_gatt.h"
+#include "beacon.h"
+#include "foundation.h"
+#include "access.h"
+#include "proxy.h"
+#include "gatt_cli.h"
+#include "proxy_msg.h"
+
+static struct {
+	const uint8_t *target;
+	struct bt_mesh_proxy_role *srv;
+} server;
+
+static void pb_gatt_msg_recv(struct bt_mesh_proxy_role *role)
+{
+	switch (role->msg_type) {
+	case BT_MESH_PROXY_PROV:
+		BT_DBG("Mesh Provisioning PDU");
+		bt_mesh_pb_gatt_recv(role->conn, &role->buf);
+		break;
+	default:
+		BT_WARN("Unhandled Message Type 0x%02x", role->msg_type);
+		break;
+	}
+}
+
+static void pb_gatt_connected(struct bt_conn *conn, void *user_data)
+{
+	server.srv = bt_mesh_proxy_role_setup(conn, bt_mesh_gatt_send,
+					      pb_gatt_msg_recv);
+
+	server.target = NULL;
```

Comment:
> `bt_mesh_pb_gatt_client_setup ` didn't reset uuid

when connection established, server.target auto cleared.

Reply:
@PavelVPV

Reply:
Can't it be cleared after right calling `bt_mesh_gatt_cli_connect`?

Reply:
I hope to be able to retry after the connection fails.